### PR TITLE
docs: simplify machine-id getting started

### PR DESCRIPTION
**Pull request by stevenGravy**
Removed separation of cloud vs self-hosted.  As a getting started wasn't much value or diff.  This was referring to parameters like `--auth-server` that weren't used that can cause confusion getting started.

### CODE REVIEW

File: docs/pages/enroll-resources/machine-id/getting-started.mdx [hunk 1]
```diff
@@ -49,17 +49,8 @@ Before you create a bot user, you need to determine which role(s) you want to
 assign to it. You can use the `tctl` command below to examine what roles exist
 on your system.
 
-<Tabs>
```
Comment (strideynet) [inline docs/pages/enroll-resources/machine-id/getting-started.mdx:52]: 👍  Yeah - this is much simpler. I don't know why we ever had this split....

**Review by strideynet — APPROVED** [review]

**Review by strideynet — COMMENTED** [review]
Looks good - thank you. Part of me is thinking we should do another PR to simplify this guide further by removing the references to `iam` joining - this made sense when we only had IAM joining, but now we have a full range of join methods, it doesn't make any sense to me to pick out AWS as the special case to include in this getting started guide.

**Comment by stevenGravy** [comment]
> Looks good - thank you. Part of me is thinking we should do another PR to simplify this guide further by removing the references to `iam` joining - this made sense when we only had IAM joining, but now we have a full range of join methods, it doesn't make any sense to me to pick out AWS as the special case to include in this getting started guide.

I did a brief to explain why you could want a non-token method. Probably worth keeping that then. pls take a look

**Comment by strideynet** [comment]
> I did a brief to explain why you could want a non-token method. Probably worth keeping that then. pls take a look

Actually - if you drop that last commit - I figured it made sense for me to raise a PR with a few changes in regards to talking about joining in this guide: https://github.com/gravitational/teleport/pull/47554

**Comment by stevenGravy** [comment]
> > I did a brief to explain why you could want a non-token method. Probably worth keeping that then. pls take a look
> 
> Actually - if you drop that last commit - I figured it made sense for me to raise a PR with a few changes in regards to talking about joining in this guide: #47554

Sure, dropped!

**Review by scottpgallagher — APPROVED** [review]
Approved.

**Pull request merged**